### PR TITLE
fix #139, #164 temp tag fix

### PR DESCRIPTION
**Pull request by KostyaSha**
i also have strategy changes and some jelly fixes, but they will be with portion of choosable strategy.

### CODE REVIEW

**Comment by KostyaSha** [comment]
Tested locally manually, removed coding style cleanups as much as i can, but not sure it good idea wasting time with reverting some coding style cleanups. 
@thomassuckow @magnayn please review.

**Comment by KostyaSha** [comment]
Please ignore last 3 commits, they are still dirty... Trying to split changes to multiple commits

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java [hunk 1]
```diff
@@ -317,8 +327,28 @@ public DockerTemplate getTemplate(Label label) {
...
         t.parent = this;
+        templates.add(t);
+    }
+
+    /**
+     * Multiple amis can have the same label.
+     * @return Templates matched to requested label assuming slave Mode
+     */
+    public List<DockerTemplate> getTemplates(Label label) {
+        ArrayList<DockerTemplate> dockerTemplates = new ArrayList<DockerTemplate>();
```
Comment (magnayn) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java]: You could use Collections2.filter() if you wanted

**Comment by magnayn** [comment]
LGTM..

**Pull request merged**